src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -103,16 +131,70 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
             passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName
             );
 
-            //
             // now, prompt for the password
-            //
-            WriteToConsole(passwordPrompt, true);
-            password = ReadLineAsSecureString();
-            if (password == null)
+
+            do
             {
-                return null;
+                WriteToConsole(passwordPrompt, true);
+                password = ReadLineAsSecureString();
+                if (password == null)
+                {
+                    return null;
+                }
             }
+            while (password.Length == 0);
+
+            if (reenterPassword)
+            {
+                reenterPasswordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_ReenterPassword, userName);
+
+
+                // now, prompt to re-enter the password.
+
+                do
+                {
+                    WriteToConsole(reenterPasswordPrompt, true);
+                    confirmPassword = ReadLineAsSecureString();
+                    if (confirmPassword == null)
+                    {
+                        return null;
+                    }
+                }
+                while (confirmPassword.Length == 0);
 
+                IntPtr pwd_ptr = IntPtr.Zero;
+                IntPtr confirmPwd_ptr = IntPtr.Zero;
+
+                try
+                {
+                    pwd_ptr = Marshal.SecureStringToBSTR(password);
+                    confirmPwd_ptr = Marshal.SecureStringToBSTR(confirmPassword);
+
+                    string pwd = Marshal.PtrToStringBSTR(pwd_ptr);
+                    string confirmPwd = Marshal.PtrToStringBSTR(confirmPwd_ptr);
+
+                    // If the string is not equal print error message and returns null.
+
+                    if (!(pwd.Equals(confirmPwd)))
+                    {
+                        reenterPasswordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PasswordMismatch);
+                        WriteToConsole(reenterPasswordPrompt, true);
+                        return null;
+                    }
```

Comment:
I'd expect a cycle here too - if user enter wrong confirmation we should prompt again for a password.

Reply:
@iSazonov  I have made the changes. Please review
``` 
Get-Credential -UserName user -ReEnterPassword

PowerShell credential request
Enter your credentials.
Password for user user: **
Re-Enter Password for user user:
Passwords do not match. Re-Enter Password for user user:
Passwords do not match. Re-Enter Password for user user:
Passwords do not match. Re-Enter Password for user user:
Passwords do not match. Re-Enter Password for user user:**
Passwords do not match. Re-Enter Password for user user:
Passwords do not match. Re-Enter Password for user user:
Passwords do not match. Re-Enter Password for user user:**


UserName                     Password
--------                     --------
user     System.Security.SecureString